lib/codeowners/checker.rb

```diff
@@ -47,7 +47,7 @@ def patterns_by_owner
         codeowners.each_with_object(hash_of_arrays) do |line, patterns_by_owner|
           next unless line.pattern?
 
-          line.owners.each { |owner| patterns_by_owner[owner] << line.pattern }
+          line.owners.each { |owner| patterns_by_owner[owner] << line.pattern.gsub(%r{^/}, '') }
```

Comment:
The initial `/` has a meaning, is it possible filter would be showing files in subdirectories even if it should not?

An example would be `/README.md`, maybe we should add a test to ensure this works properly?

Reply:
With our current implementation using ruby-git it always searches from the root folder without `/` and with it, it fails. It will not show files in subdirectories but not even for the patterns without `/`. The same problem is with the `check` method.

I don't think it is the correct behavior, we would probably have to fix the original ruby-git implementation of the pattern matching.

Talking about these two lines: 
https://github.com/toptal/codeowners-checker/blob/0373e27da61c8c19682281ed1ee56d425a08f75c/lib/codeowners/checker.rb#L42
https://github.com/toptal/codeowners-checker/blob/0373e27da61c8c19682281ed1ee56d425a08f75c/lib/codeowners/checker.rb#L82

Reply:
You can try: `pattern.gsub(%r{^/}, Dir.pwd+'/')` it should replace the pattern and it is consistent with the proper syntax of gitignore which matches only paths starting from root of the repository.

Reply:
Yes, this works and it matches files from root but it is no difference to files without the slash, that was my point. Files from subdirectories will never be shown.
* with / and `gsub(%r{^/}, Dir.pwd+'/')` - matches from the root
* without / - matches from the root
 
It doesn't seem right, if the slash has a meaning, there should be a difference.

Reply:
we decided to move the discussion to another issue